addons/binding/org.openhab.binding.feican/src/main/java/org/openhab/binding/feican/handler/FeicanHandler.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.feican.handler;
+
+import static org.openhab.binding.feican.FeicanBindingConstants.*;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.feican.internal.Commands;
+import org.openhab.binding.feican.internal.Connection;
+import org.openhab.binding.feican.internal.FeicanConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FeicanHandler} is responsible for handling commands, which are sent to one of the channels.
+ *
+ * @author Hilbrand Bouwkamp - Initial contribution
+ */
+@NonNullByDefault
+public class FeicanHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FeicanHandler.class);
+    private final Commands commands = new Commands();
+
+    @Nullable
+    private Connection connection;
+
+    /**
+     * Creates a new handler for the Feican thing.
+     *
+     * @param thing The thing to create the handler for
+     */
+    public FeicanHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof OnOffType) {
+                handleSwithOnOff((OnOffType) command);
+            } else if (command instanceof HSBType) {
+                handleColor(channelUID, (HSBType) command);
+            } else if (command instanceof PercentType) {
+                handlePercentage(channelUID, (PercentType) command);
+            } else if (command instanceof StringType) {
+                handleString(channelUID, (StringType) command);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
```

Comment:
What about the REFRESH command? Or did you leave it out, because you cannot get any state updates from the bulb?

Reply:
yes I left it out for the reason I can't get the status from the device.